### ISSUE TITLE
`TypeError` if a global decorator is used on the class method.

### ISSUE THREAD

**Issue by TerenceZ** [babel/babel-eslint#229]
After updating to `babel@6.3.26` and `babel-eslint@5.0.0-beta6`,  if I lint the code sample below using  the followed `.eslint` config, it will raise up a type error. ~~But if I use `babel@6.3.13` and `babel-eslint@4.1.6`, it will be ok!~~

By the way, the `eslint` version is 1.10.3.

```js
// Code sample
// If we use the decorator on the class, it will be ok, but if we
// use it on the class method, it will be wrong for `babel-eslint@5.0.0-beta6`.
export default class A {
  @test
  f() {

  }
}
```

```js
// .eslintrc
{
  "parser": "babel-eslint",
  "globals": {
    "test": true
  },
  "rules": {
    "no-undef": 2
  }
}
```

Here is the error:

```js
[TypeError: Cannot read property 'type' of undefined]
TypeError: Cannot read property 'type' of undefined
    at hasTypeOfOperator (node_modules/eslint/lib/rules/no-undef.js:56:18)
    at node_modules/eslint/lib/rules/no-undef.js:81:21
    at Array.forEach (native)
    at EventEmitter.Program:exit (node_modules/eslint/lib/rules/no-undef.js:77:33)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.leaveNode (node_modules/eslint/lib/util/node-event-generator.js:51:22)
    at CommentEventGenerator.leaveNode (node_modules/eslint/lib/util/comment-event-generator.js:111:23)
    at Controller.controller.traverse.leave (node_modules/eslint/lib/eslint.js:770:36)
    at Controller.__execute (node_modules/estraverse/estraverse.js:397:31)
```

**Comment by taverasmisael** [comment]
:+1:

**Comment by christophehurpeau** [comment]
:+1:

**Comment by adriantoine** [comment]
Same here :+1:

**Comment by danez** [comment]
I just tested and this was fixed in 6.0.4 by #296
